crypto/rsa/rsa_pmeth.c

```diff
@@ -144,9 +144,14 @@ static int pkey_rsa_sign(EVP_PKEY_CTX *ctx, unsigned char *sig,
      * be reflected back in the "original" key.
      */
     RSA *rsa = (RSA *)EVP_PKEY_get0_RSA(ctx->pkey);
+    int md_size;
 
     if (rctx->md) {
-        if (tbslen != (size_t)EVP_MD_get_size(rctx->md)) {
+        md_size = EVP_MD_get_size(rctx->md);
+        if (md_size <= 0)
+            return -1;
```

Comment:
Please join this with the following if as I think raising RSA_R_INVALID_DIGEST_LENGTH is appropriate here too.